ext/puma_http11/org/jruby/puma/MiniSSL.java

```diff
@@ -47,6 +65,36 @@ public static void createMiniSSL(Ruby runtime) {
     RubyModule mPuma = runtime.defineModule("Puma");
     RubyModule ssl = mPuma.defineModuleUnder("MiniSSL");
 
+    /**
+     * Define constants to match parity with MRI OpenSSL extention for Puma.
+     */
+
+    // this constant is only used for testing; lets set it to true.
+    // java by default does not have SSL3/TLS1 enabled; and developers have to go through hoops
+    // to turn it on.
+    ssl.defineConstant("OPENSSL_NO_SSL3", runtime.newBoolean(true));
+    ssl.defineConstant("OPENSSL_NO_TLS1", runtime.newBoolean(true));
+
+    if (!Boolean.getBoolean(NETTY_USE_KEY)) {
+      ssl.defineConstant("OPENSSL_LIBRARY_VERSION", runtime.newString("Unknown"));
+      ssl.defineConstant("OPENSSL_VERSION", runtime.newString("Unknown"));
+    } else {
+      try {
+        ssl.defineConstant(
+                "OPENSSL_LIBRARY_VERSION",
+                runtime.newString(io.netty.handler.ssl.OpenSsl.versionString())
+        );
+        ssl.defineConstant(
+                "OPENSSL_VERSION",
+                runtime.newString(io.netty.handler.ssl.OpenSsl.versionString())
+        );
+      } catch (Throwable t) {
+        LOGGER.log(Level.INFO, "Failed to use Netty OpenSSL " + t.getMessage(), t);
+        ssl.defineConstant("OPENSSL_LIBRARY_VERSION", runtime.newString("Unknown"));
+        ssl.defineConstant("OPENSSL_VERSION", runtime.newString("Unknown"));
+      }
+    }
```

Comment:
These constants were missing which would actually cause the test file **test_puma_server_ssl.rb** to actually skip itself for JRuby because _DISABLE_SSL_ was true due to the missing constants.

We now have test coverage again for JRuby + SSL !
🍾 🎆 

> rake test TEST=test/test_puma_server_ssl.rb
> Fabulous run in 0.895517s, 12.2834 runs/s, 10.0501 assertions/s.
> 11 runs, 9 assertions, 0 failures, 0 errors, 2 skips

Reply:
Thanks for adding this!